lib/ash_json_api/json_schema/open_api_schema.ex

```diff
@@ -4,190 +4,248 @@ defmodule AshJsonApi.OpenApiSchema do
 
   @doc "See https://swagger.io/specification/#openapi-object"
   def generate(api) do
-    resources =
-      api
-      |> Ash.Api.Info.resources()
-      |> Enum.filter(&(AshJsonApi.Resource in Spark.extensions(&1)))
-
-    paths =
-      Enum.flat_map(resources, fn resource ->
-        resource
-        |> AshJsonApi.Resource.Info.routes()
-        |> Enum.map(&open_api_path(&1, api, resource))
-      end)
-      |> Enum.group_by(fn {path, _path_item} -> path end, fn {_path, path_item} -> path_item end)
-      |> Map.new(fn {path, path_items} -> {path, Map.new(path_items)} end)
-
-    definitions =
-      Enum.reduce(resources, JsonSchema.base_definitions(definitions_path: "#/components/schemas"), fn resource, acc ->
-        Map.put(acc, AshJsonApi.Resource.Info.type(resource), JsonSchema.resource_object_schema(resource))
-      end)
-
-    info = openapi_info(api)
-
-    common_responses = %{
-      "errors" => %{
-        "description" => "General Error",
-        "content" => %{
-          "application/vnd.api+json" => %{
-            "schema" => %{
-              "$ref" => "#/components/schemas/errors"
-            }
-          }
-        }
-      }
-    }
+    resources = resources(api)
 
     %{
       # Note: Using version 3.0.3 as some tooling such as SwaggerUI hasn't updated to 3.1.0 yet.
       "openapi" => "3.0.3",
-      "info" => info,
-      "paths" => paths,
+      "info" => info(api),
+      "paths" => paths(api, resources),
       "components" => %{
-        "responses" => common_responses,
-        "schemas" => definitions
+        "responses" => responses(),
+        "schemas" => schemas(resources)
       },
       "servers" => [],
       "security" => [],
-      "tags" => []
+      "tags" => tags(resources)
       # "externalDocs" => nil,
       # "extensions" => nil
     }
   end
 
   @doc "See https://swagger.io/specification/#info-object"
-  def openapi_info(_api) do
+  def info(api) do
     %{
-      "title" => "title pending",
-      "description" => "description pending",
-      "version" => "api document version pending"
+      "title" => String.capitalize("#{Ash.Api.Info.short_name(api)}") <> " API",
+      "description" => description(api),
+      "version" => document_version()
     }
   end
 
-  def open_api_path(%{method: :get} = route, api, resource) do
-    method = "get"
-    {href, path_params} = JsonSchema.route_href(route, api)
+  def description(api) do
+    # TODO: Get this value from a DSL attribute
+    case Code.fetch_docs(api) do
```

Comment:
Yeah, we should add

```elixir
api do
  description "..."
end
```

to the API dsl. But we definitely can't expose the code docs in the actual released version so we'll want to add the above first.